public/js/app/controllers.js

```diff
@@ -38,9 +38,25 @@ var ctrl = angular.module("mockapp.controllers",['mockapp.services','mockapp.fac
             $scope.dropdown = function() {
               if ($scope.sutChecked == false){
                   $scope.sutlist = sutService.getAllSUT();
+                  $scope.groupMessage = "";
                }
             };
 
+            $scope.checkDuplicateGroup = function (){
+              count=0;
+              $scope.groupMessage = "";
+              if($scope.sutChecked == true){
+                for(i=0; i<$scope.sutlist.length;i++){
```

Comment:
Since the variable `i` is not defined with `let` or `var`, it will unexpectedly default to global scope.

Please change this to `for (var i=0; i<$scope.sutlist.length; i++)` to scope `i` correctly.

Please do this on lines 743 and 805 as well.